Docker build fails

## Type of Issue

Bug (eventually)

## Background info

I would like to run excalidraw locally (MacOS) using Docker.
The build fails with an error that suggests that either there is not enough memory or someone explicitly killed the process.
I can exclude both reasons, since I am working alone on a fresh MacBook with 32gb of RAM and a lot of free disk space.

## Log

```
❯ cd Desktop
❯ git clone https://github.com/excalidraw/excalidraw.git
Cloning into 'excalidraw'...
remote: Enumerating objects: 308, done.
remote: Counting objects: 100% (308/308), done.
remote: Compressing objects: 100% (224/224), done.
remote: Total 12711 (delta 195), reused 115 (delta 84), pack-reused 12403
Receiving objects: 100% (12711/12711), 18.14 MiB | 7.43 MiB/s, done.
Resolving deltas: 100% (9817/9817), done.
❯ cd excalidraw
❯ docker build -t excalidraw/excalidraw .
[+] Building 30.5s (14/15)
 => [internal] load build definition from Dockerfile                                                                     0.0s
 => => transferring dockerfile: 407B                                                                                     0.0s
 => [internal] load .dockerignore                                                                                        0.0s
 => => transferring context: 148B                                                                                        0.0s
 => [internal] load metadata for docker.io/library/nginx:1.17-alpine                                                     1.4s
 => [internal] load metadata for docker.io/library/node:14-alpine                                                        1.4s
 => [auth] library/node:pull token for registry-1.docker.io                                                              0.0s
 => [auth] library/nginx:pull token for registry-1.docker.io                                                             0.0s
 => [build 1/6] FROM docker.io/library/node:14-alpine@sha256:b2da3316acdc2bec442190a1fe10dc094e7ba4121d029cb32075ff59bb  0.0s
 => [internal] load build context                                                                                        0.1s
 => => transferring context: 4.18MB                                                                                      0.1s
 => CACHED [stage-1 1/2] FROM docker.io/library/nginx:1.17-alpine@sha256:763e7f0188e378fef0c761854552c70bbd817555dc4de0  0.0s
 => CACHED [build 2/6] WORKDIR /opt/node_app                                                                             0.0s
 => CACHED [build 3/6] COPY package.json package-lock.json ./                                                            0.0s
 => CACHED [build 4/6] RUN npm i --no-optional                                                                           0.0s
 => CACHED [build 5/6] COPY . .                                                                                          0.0s
 => ERROR [build 6/6] RUN npm run build:app:docker                                                                      28.9s
------
 > [build 6/6] RUN npm run build:app:docker:
#14 0.653
#14 0.653 > excalidraw@ build:app:docker /opt/node_app
#14 0.653 > REACT_APP_INCLUDE_GTAG=false REACT_APP_DISABLE_SENTRY=true react-scripts build
#14 0.653
#14 3.158 Creating an optimized production build...
#14 28.77 The build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process.
#14 28.78 npm ERR! code ELIFECYCLE
#14 28.78 npm ERR! errno 1
#14 28.79 npm ERR! excalidraw@ build:app:docker: `REACT_APP_INCLUDE_GTAG=false REACT_APP_DISABLE_SENTRY=true react-scripts build`
#14 28.79 npm ERR! Exit status 1
#14 28.79 npm ERR!
#14 28.79 npm ERR! Failed at the excalidraw@ build:app:docker script.
#14 28.79 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
#14 28.80
#14 28.80 npm ERR! A complete log of this run can be found in:
#14 28.80 npm ERR!     /root/.npm/_logs/2020-12-29T12_33_15_238Z-debug.log
------
executor failed running [/bin/sh -c npm run build:app:docker]: exit code: 1
❯ npx sb info

Environment Info:

  System:
    OS: macOS 11.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 12.14.1 - ~/.nvm/versions/node/v12.14.1/bin/node
    Yarn: 1.21.1 - ~/.nvm/versions/node/v12.14.1/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.14.1/bin/npm
  Browsers:
    Chrome: 87.0.4280.88
    Firefox: 84.0
    Safari: 14.0.2

```

## Findings 
- The required Node Version (12) matches the one used during the installation
- Other reports of the same error (https://github.com/facebook/create-react-app/issues/3862, https://github.com/vercel/vercel/issues/1080) suggest adding more memory, but I can hardly imagine that 32Gb should not be sufficient
- I closed all memory-extensive apps like webstorm

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,10 +1,10 @@
 *
+!.env
+!.eslintrc.json
+!.npmrc
+!.prettierrc
+!package.json
 !public/
 !src/
-!.npmrc
-!.eslintrc.json
-!.prettierrc
-!package-lock.json
-!package.json
 !tsconfig.json
-!.env
+!yarn.lock

--- a/.github/workflows/build-docker.yml
+++ b/.github/workflows/build-docker.yml
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+  pull_request:
 
 jobs:
   build:

--- a/.github/workflows/build-docker.yml
+++ b/.github/workflows/build-docker.yml
@@ -4,7 +4,6 @@ on:
   push:
     branches:
       - master
-  pull_request:
 
 jobs:
   build:

--- a/.github/workflows/build-docker.yml
+++ b/.github/workflows/build-docker.yml
@@ -4,11 +4,10 @@ on:
   push:
     branches:
       - master
-  pull_request:
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - run: docker build -t excalidraw .

--- a/.github/workflows/build-packages.yml
+++ b/.github/workflows/build-packages.yml
@@ -11,9 +11,9 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - name: Setup Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: 14.x
       - name: Install dependencies

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -7,10 +7,10 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
 
       - name: Setup Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: 14.x
 

--- a/.github/workflows/locales-coverage.yml
+++ b/.github/workflows/locales-coverage.yml
@@ -15,7 +15,7 @@ jobs:
           token: ${{ secrets.PUSH_TRANSLATIONS_COVERAGE_PAT }}
 
       - name: Setup Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: 14.x
 

--- a/.github/workflows/sentry-production.yml
+++ b/.github/workflows/sentry-production.yml
@@ -9,9 +9,9 @@ jobs:
   release:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1.0.0
+      - uses: actions/checkout@v2
       - name: Setup Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: 14.x
       - name: Install and build

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -6,9 +6,9 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - name: Setup Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: 14.x
       - name: Install and test

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -18,7 +18,7 @@ services:
     volumes:
       - ./:/opt/node_app/app:delegated
       - ./package.json:/opt/node_app/package.json
-      - ./package-lock.json:/opt/node_app/package-lock.json
+      - ./yarn.lock:/opt/node_app/yarn.lock
       - notused:/opt/node_app/app/node_modules
 
 volumes:

